crates/oxc_transformer/src/plugins/module_runner_transform.rs

```diff
@@ -168,18 +171,10 @@ impl<'a> ModuleRunnerTransform<'a> {
                     new_stmts.push(stmt);
                 }
             }
-
-            // Found that non-import statements, we should start hoisting import statements after this
-            if hoist_index.is_none() {
-                hoist_index.replace(new_stmts.len() - 1);
-            }
         }
 
-        if let Some(index) = hoist_index {
-            if !hoist_imports.is_empty() {
-                new_stmts.splice(index..index, hoist_imports);
-            }
-        }
+        new_stmts.splice(0..0, hoist_imports);
+        new_stmts.splice(0..0, hoist_exports);
```

Comment:
```suggestion
        new_stmts.splice(0..0, hoist_exports.into_iter().chain(hoist_imports);
```

This looks like it would work better - What do you think?

Reply:
It works! I don't understand the perf difference though :smile: